kafka/producer.go

```diff
@@ -128,7 +121,7 @@ func (p *Producer) produce(msg *Message, msgFlags int, deliveryChan chan Event)
 	cErr := C.do_produce(p.handle.rk, crkt,
 		C.int32_t(msg.TopicPartition.Partition),
 		C.int(msgFlags)|C.RD_KAFKA_MSG_F_COPY,
-		unsafe.Pointer(valp), C.size_t(valLen),
+		C.CBytes(msg.Value), C.size_t(valLen),
```

Comment:
CBytes makes a copy of the payload, which we would want to avoid, or handle by using MSG_F_FREE.

It also requires Golang >= 1.7, a limitation we don't want to enforce yet.